docs/website/docs/plus/features/project/source-configuration.md

```diff
@@ -0,0 +1,133 @@
+
+
+```yaml
+sources:
+  pokemon_api:
+    type: rest_api
+    client:
+      base_url: https://pokeapi.co/api/v2/
+      paginator: auto
+    resource_defaults:
+      primary_key: name
+    resources:
+      - pokemon
+      - berry
+      - 
+        name: encounter_conditions
+        endpoint:
+          path: encounter-condition
+          params:
+            offset:
+              type: incremental
+              cursor_path: name
+        write_disposition: append
+```
+
+* `type: rest_api`: Specifies the use of the built-in REST API source.
+* `client.base_url`: Sets the root URL for all API requests.
+* `paginator: auto`: Enables automatic detection and handling of pagination.
+* `resource_defaults`: Contains the default values to configure the dlt resources. This configuration is applied to all resources unless overridden by the resource-specific configuration.
+* Each item in `resources`defines an endpoint to extract. Simple entries like `pokemon`and `berry` will fetch from `/pokemon` and `/berry`, respectively.
+* The `encounter-condition` resource uses an advanced configuration:
+  * `path`: Point to the `/encounter-condition`endpoint.
+  * `params.offset`: Enables incremental loading using the `name` field as the cursor.
+  * `write_disposition: append`: Ensures new data is appended rather than overwriting previous loads.
+
+
+
+## SQL database
+
+For SQL-base extractions that require no table-specific parameter configuration, it's possible to initialize `type = sql_database` and declare multiple tables at once.
+ 
+### General SQL Database Source
+
+```yaml 
+sources:
+   sql_source:
+    type: sql_database
+    table_names: 
+      - family
+      - clan
+    incremental:
+        cursor_path: updated
+        initial_value: 2023-01-12T11:21:28Z 
+```
+
+This defines a connection to a SQL database with incremental loading applied across multiple tables.
+
+* `type: sql_database`: Specifies the SQL database connector.
+
+* `table_names`: List of tables to extract.
+
+* `incremental`: Global configuration for incremental extraction.
+
+### Table-Specific Configuration
+
+For table specific configurationssettings such as different `primary_key`s, individual tables can be defined as standalone sources using the `sql_table` type.
+
+
+```yaml
+sources: 
+  sql_family:
+      type: sql_database.sql_table
+      table: family
+      incremental:
+        cursor_path: updated
+        initial_value: 2023-01-12T11:21:28Z 
+      primary_key: rfam_id
+```
+
+* `type: sql_table`: Indicates a single-table extraction.
+
+* `table`: Name of the table to extract.
+
+* `incremental`: Enables incremental loading for the table.
+
+* `primary_key`: Specifies the table's unique identifier for deduplication and merges.
+
+
+## Filesystem
+
+Filesystem sources can be set via the `readers` type and the filesystem specific resources can be called via the CLI `run pipline` command.
+
+
+```yaml
+sources: 
+ file_source:
+    type: filesystem.readers
```

Comment:
Shouldn't it be 
```suggestion
    type: filesystem.readers.read_csv
```
?
Makes sense to define it here once instead of calling the --resources on each run